owncloudComLibrary/src/test/java/com/owncloud/android/lib/resources/shares/responses/ShareeResponseTest.kt

```diff
@@ -0,0 +1,128 @@
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in
+ *   all copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
+ */
+
+package com.owncloud.android.lib.resources.shares.responses
+
+import com.owncloud.android.lib.resources.CommonOcsResponse
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import java.io.File
+import java.lang.reflect.Type
+
+class ShareeResponseTest {
+
+    lateinit var response: CommonOcsResponse<ShareeOcsResponse>
+
+    private fun loadResponses(fileName: String, adapter: JsonAdapter<CommonOcsResponse<ShareeOcsResponse>>) =
+        adapter.fromJson(File(fileName).readText())
+
+    @Before
+    fun prepare() {
+        val moshi = Moshi.Builder().build()
+        val type: Type = Types.newParameterizedType(CommonOcsResponse::class.java, ShareeOcsResponse::class.java)
+        val adapter: JsonAdapter<CommonOcsResponse<ShareeOcsResponse>> = moshi.adapter(type)
+        response = loadResponses(EXAMPLE_RESPONSE_JSON, adapter)!!
+    }
+
+    @Test
+    fun `check structure - ok - example response files exist`() {
+        val file = File(EXAMPLE_RESPONSE_JSON)
+        assertTrue(file.exists())
+    }
+
+    @Test
+    fun `check structure - ok - contains meta`() {
+        assertEquals("OK", response.ocs.meta.message!!)
+        assertEquals(200, response.ocs.meta.statusCode!!)
+        assertEquals("ok", response.ocs.meta.status!!)
+        assertTrue(response.ocs.meta.itemsPerPage?.isEmpty()!!)
+        assertTrue(response.ocs.meta.totalItems?.isEmpty()!!)
+    }
+
+    @Test
+    fun `check structure - ok - contains exact`() {
+        assertNotEquals(null, response.ocs.data.exact)
+    }
+
+    @Test
+    fun `check structure - ok - contains groups`() {
+        assertNotEquals(null, response.ocs.data.groups)
+    }
+
+    @Test
+    fun `check structure - ok - contains remotes`() {
+        assertNotEquals(null, response.ocs.data.remotes)
+    }
+
+    @Test
+    fun `check structure - ok - contains users`() {
+        assertNotEquals(null, response.ocs.data.users)
```

Comment:
If we are using the same response on each test, why shouldn't we unify them in a single one?

`assertNull`
`assertNotNull` 🤔

Reply:
Please correct me if I am wrong, but I learned that every test should test one thing and not everything. Am I wrong?